File: MovieHostingLinkGripper/src/main/java/com/sheryv/tools/movielinkgripper/search/SearchResult.java

package com.sheryv.tools.movielinkgripper.search;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class SearchResult {

    private long page;
    private List<SearchItem> results;
    @JsonProperty("total_pages")
    private long totalPages;
    @JsonProperty("total_results")
    private long totalResults;

}

File: CmdTools/src/main/kotlin/com/sheryv/tools/cmd/convertmovienames/videosearch/SearchResult.kt
package com.sheryv.tools.cmd.convertmovienames.videosearch

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class SearchResult<R : SearchItem>(
  val page: Long = 0,
  val results: List<R>,
  @JsonProperty("total_pages")
  val totalPages: Long,
  @JsonProperty("total_results")
  val totalResults: Long,
) {
}
